x/ecocredit/base/simulation/operations.go

```diff
@@ -244,12 +293,34 @@ func WeightedOperations(
 			weightMsgBridge,
 			SimulateMsgBridge(ak, bk, qryClient),
 		),
+		simulation.NewWeightedOperation(
+			weightMsgAddCreditType,
+			SimulateMsgAddCreditType(ak, bk, govk, qryClient, authority),
+		),
+
+		simulation.NewWeightedOperation(
+			weightMsgAddClassCreator,
+			SimulateMsgAddClassCreator(ak, bk, govk, qryClient, authority),
+		),
+
+		simulation.NewWeightedOperation(
+			weightMsgRemoveClassCreator,
+			SimulateMsgRemoveClassCreator(ak, bk, govk, qryClient, authority),
+		),
+
+		simulation.NewWeightedOperation(
+			weightMsgToggleClassAllowlist,
+			SimulateMsgToggleClassAllowlist(ak, bk, govk, qryClient, authority),
+		),
+		simulation.NewWeightedOperation(
+			weightMsgUpdateClassFees,
+			SimulateMsgUpdateClassFees(ak, bk, govk, qryClient, authority),
+		),
 	}
 
-	// TODO: #1363
-	// basketOps := basketsims.WeightedOperations(appParams, cdc, ak, bk, qryClient, basketQryClient)
-	// ops = append(ops, basketOps...)
-	marketplaceOps := marketsims.WeightedOperations(appParams, cdc, ak, bk, qryClient, mktQryClient)
+	basketOps := basketsims.WeightedOperations(appParams, cdc, ak, bk, govk, qryClient, basketQryClient, authority)
+	ops = append(ops, basketOps...)
+	marketplaceOps := marketsims.WeightedOperations(appParams, cdc, ak, bk, qryClient, mktQryClient, govk, authority)
 
 	return append(ops, marketplaceOps...)
```

Comment:
Ok for now but I'm wondering if we can somehow move this to the module `WeightedOperations` method:

```
// WeightedOperations implements AppModuleSimulation/WeightedOperations.
func (m Module) WeightedOperations(simState module.SimulationState) []simtypes.WeightedOperation {
	baseServer, basketServer, marketServer := m.Keeper.QueryServers()
	return basesims.WeightedOperations(
		simState.AppParams,
		simState.Cdc,
		m.accountKeeper,
		m.bankKeeper,
		baseServer,
		basketServer,
		marketServer,
	)
}
```

Reply:
Maybe an idea for the future but ideally we wouldn't need to import basket and marketplace operations into the operations for the base package/submodule. Might also be more trouble than its worth.